### ISSUE TITLE
qvm-template - strange interaction with dnf

### ISSUE THREAD

**Issue by unman** [QubesOS/qubes-issues#7252]
[How to file a helpful issue](https://www.qubes-os.org/doc/issue-tracking/)

### Qubes OS release
4.1 clean install and fully updated.

### Brief summary
qvm-template has strange and confusing interaction with standard dnf tool in dom0.

### Steps to reproduce
These are not steps, but separate use cases:

1. Enable qubes-templates-itl-testing repository in /etc/yum.repos.d/qubes-templates.repo
2. Install template with dnf install.
3. Install template with qvm-template tool.

### Expected behavior
The output of qvm-template will be consistent with dnf configuration, and vice versa

### Actual behavior
In Case 1: `qvm-template list` only shows available templates from qubes-templates-itl, although that repository is enabled.
After #7170, it is possible to view the contents of qubes-templates-itl-testing , but this must be explicitly specified as an option to qvm-template.

Case 2: Templates installed with dnf do not appear in the output of `qvm-template`.

Case 3: Templates installed with qvm-template do not appear in the output of `dnf list installed`.

All of these are unexpected and confusing.

**Comment by marmarek** [comment]
dnf should no longer be used to manage templates - we added redirection from qubes-dom0-update (which is/was the documented interface for installing dom0 packages), but not dnf itself.

I think the source of most of those confusion is that `/etc/yum.repos.d/qubes-template.repo` is still present - it shouldn't be. The thing that is actually used is `/etc/qubes/repo-templates/qubes-templates.repo`.

Thanks for the report, we clearly forgot about this change too.

**Comment by adrelanos** [comment]
Whonix also affected.

qubes-templates-community-testing repo is enabled.

    qvm-template list

```
Installed Templates
debian-11          0:4.0.6-202201171556  @commandline
debian-11-minimal  0:4.0.6-202108191622  qubes-templates-itl
fedora-34          0:4.0.6-202201160401  @commandline
whonix-gw-16       0:4.0.6-202201160402  @commandline
whonix-ws-16       0:4.0.6-202201160402  @commandline
Available Templates
debian-10-minimal  0:4.0.6-202010131933  qubes-templates-itl
debian-10          0:4.0.6-202009131420  qubes-templates-itl
debian-11-minimal  0:4.0.6-202108191622  qubes-templates-itl
debian-11          0:4.0.6-202201171556  qubes-templates-itl
fedora-32-minimal  0:4.0.6-202010191916  qubes-templates-itl
fedora-32-xfce     0:4.0.6-202010191916  qubes-templates-itl
fedora-32          0:4.0.6-202010192324  qubes-templates-itl
fedora-33-minimal  0:4.0.6-202102131637  qubes-templates-itl
fedora-33-xfce     0:4.0.6-202102140100  qubes-templates-itl
fedora-33          0:4.0.6-202102131637  qubes-templates-itl
fedora-34-minimal  0:4.0.6-202110020922  qubes-templates-itl
fedora-34-xfce     0:4.0.6-202110020922  qubes-templates-itl
fedora-34          0:4.0.6-202201160401  qubes-templates-itl
```

But Whonix templates aren't listed.

Requires using `--enablerepo=qubes-templates-community-testing` parameter but that should not be required.

    qvm-template list --enablerepo=qubes-templates-community-testing

```
Installed Templates
debian-11          0:4.0.6-202201171556  @commandline
debian-11-minimal  0:4.0.6-202108191622  qubes-templates-itl
fedora-34          0:4.0.6-202201160401  @commandline
whonix-gw-16       0:4.0.6-202201160402  @commandline
whonix-ws-16       0:4.0.6-202201160402  @commandline
Available Templates
centos-8-minimal   0:4.0.6-202010111524  qubes-templates-community-testing
centos-8-xfce      0:4.0.6-202009272100  qubes-templates-community-testing
centos-8           0:4.0.6-202010191306  qubes-templates-community-testing
debian-10-minimal  0:4.0.6-202010131933  qubes-templates-itl
debian-10          0:4.0.6-202009131420  qubes-templates-itl
debian-11-minimal  0:4.0.6-202108191622  qubes-templates-itl
debian-11          0:4.0.6-202201171556  qubes-templates-itl
fedora-32-minimal  0:4.0.6-202010191916  qubes-templates-itl
fedora-32-xfce     0:4.0.6-202010191916  qubes-templates-itl
fedora-32          0:4.0.6-202010192324  qubes-templates-itl
fedora-33-minimal  0:4.0.6-202102131637  qubes-templates-itl
fedora-33-xfce     0:4.0.6-202102140100  qubes-templates-itl
fedora-33          0:4.0.6-202102131637  qubes-templates-itl
fedora-34-minimal  0:4.0.6-202110020922  qubes-templates-itl
fedora-34-xfce     0:4.0.6-202110020922  qubes-templates-itl
fedora-34          0:4.0.6-202201160401  qubes-templates-itl
gentoo-minimal     0:4.0.6-202201041342  qubes-templates-community-testing
gentoo-xfce        0:4.0.6-202201041342  qubes-templates-community-testing
gentoo             0:4.0.6-202201071608  qubes-templates-community-testing
kali               0:4.0.6-202106171816  qubes-templates-community-testing
whonix-gw-15       0:4.0.6-202107061442  qubes-templates-community-testing
whonix-gw-16       0:4.0.6-202201160402  qubes-templates-community-testing
whonix-ws-15       0:4.0.6-202107061442  qubes-templates-community-testing
whonix-ws-16       0:4.0.6-202201160402  qubes-templates-community-testing
```

**Comment by adrelanos** [comment]
For the following I am not sure. Is it the same bug or a different bug?

    qvm-template list --enablerepo=qubes-templates-community

```
Installed Templates
debian-11          0:4.0.6-202201171556  @commandline
debian-11-minimal  0:4.0.6-202108191622  qubes-templates-itl
fedora-34          0:4.0.6-202201160401  @commandline
whonix-gw-16       0:4.0.6-202201160402  @commandline
whonix-ws-16       0:4.0.6-202201160402  @commandline
Available Templates
debian-10-minimal  0:4.0.6-202010131933  qubes-templates-itl
debian-10          0:4.0.6-202009131420  qubes-templates-itl
debian-11-minimal  0:4.0.6-202108191622  qubes-templates-itl
debian-11          0:4.0.6-202201171556  qubes-templates-itl
fedora-32-minimal  0:4.0.6-202010191916  qubes-templates-itl
fedora-32-xfce     0:4.0.6-202010191916  qubes-templates-itl
fedora-32          0:4.0.6-202010192324  qubes-templates-itl
fedora-33-minimal  0:4.0.6-202102131637  qubes-templates-itl
fedora-33-xfce     0:4.0.6-202102140100  qubes-templates-itl
fedora-33          0:4.0.6-202102131637  qubes-templates-itl
fedora-34-minimal  0:4.0.6-202110020922  qubes-templates-itl
fedora-34-xfce     0:4.0.6-202110020922  qubes-templates-itl
fedora-34          0:4.0.6-202201160401  qubes-templates-itl
```

Doesn't list Whonix under `Available Templates` but should already be migrated to `qubes-templates-community` as per https://github.com/QubesOS/qubes-issues/issues/6890#issuecomment-1031429983.

**Comment by marmarek** [comment]
> qubes-templates-community-testing repo is enabled.

See my comment above - different repo file is now in use for templates (`/etc/qubes/repo-templates/qubes-templates.repo`). You can also confirm with `qvm-template repolist`. I'll replace the old file with a placeholder pointing to the new one.

**Comment by lyuts** [comment]
I apologize if I'm missing something obvious, but I think @adrelanos's concern still stands. With proper repo file in place, and not having legacy template file in `/etc/yum.repos.d/`, `qvm-template list --enablerepo=qubes-templates-community` doesn't show whonix templates as being available.

**Comment by marmarek** [comment]
> I apologize if I'm missing something obvious, but I think @adrelanos's concern still stands. With proper repo file in place, and not having legacy template file in `/etc/yum.repos.d/`, `qvm-template list --enablerepo=qubes-templates-community` doesn't show whonix templates as being available.

Yes, that's separate issue.

**Comment by adrelanos** [comment]
> > I apologize if I'm missing something obvious, but I think @adrelanos's concern still stands. With proper repo file in place, and not having legacy template file in `/etc/yum.repos.d/`, `qvm-template list --enablerepo=qubes-templates-community` doesn't show whonix templates as being available.
> 
> Yes, that's separate issue.

Was this fixed here too, or has a separate ticket or a new ticket needs to be created?

**Comment by marmarek** [comment]
The issue of `qubes-templates-community` repo appearing as empty is fixed already. No new ticket is necessary anymore.

The issue about confusing configuration of template repositories (this ticket) is not fixed yet (although linked PRs should fix it).
